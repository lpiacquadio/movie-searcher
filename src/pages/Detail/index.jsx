import React from 'react'

export function Detail() {
    return <div>Detail Page</div>
}
