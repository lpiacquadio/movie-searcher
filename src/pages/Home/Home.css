.Searcher {
    background: antiquewhite;
    padding: 30px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
}

.Title {
    font-weight: bold;
    font-size: 1.5rem;
    font-family: monospace;
    cursor: default;
}

.Content {
    padding-top: 20px;
    min-height: calc(100vh - 110px);
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
}

@media screen and (min-width: 37.5rem) {
    .Title {
        font-size: 1.75rem;
    }
}

@media screen and (min-width: 60rem) {
    .Title {
        font-size: 2rem;
    }
}
