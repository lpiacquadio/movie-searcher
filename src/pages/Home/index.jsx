import React from 'react'
import { useShows } from '../../hooks/useShows'
import { Search } from '../../components/Search'
import { Spinner } from '../../components/Spinner'
import { Info } from '../../components/Info'
import { ListOfShows } from '../../components/ListOfShows'
import { DEFAULT_TEXTS } from '../../services/settings'
import './Home.css'

export function Home() {
    const { loading, shows, search, onSearch } = useShows('')
    return (
        <div>
            <header className="Searcher">
                <div className="Title">{DEFAULT_TEXTS.TITLE}</div>
                <Search
                    onSearch={onSearch}
                    placeholder={DEFAULT_TEXTS.SEARCH_PLACEHOLDER}
                    value={search}
                />
            </header>
            <main className="Content">
                {loading ? (
                    <Spinner />
                ) : (
                    <ListOfShows
                        onNotStarted={<Info>{DEFAULT_TEXTS.NOT_STARTED}</Info>}
                        onNotFound={<Info>{DEFAULT_TEXTS.NOT_FOUND}</Info>}
                        shows={shows}
                        search={search}
                    />
                )}
            </main>
        </div>
    )
}
