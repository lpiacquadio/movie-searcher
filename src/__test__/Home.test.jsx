import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, test } from 'vitest'
import { Home } from '../pages/Home'

describe('Home', () => {
    test('Home component renders correctly', () => {
        const component = renderer.create(<Home />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
