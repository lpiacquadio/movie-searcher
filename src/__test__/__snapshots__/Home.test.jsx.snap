// Vitest Snapshot v1

exports[`Home > Home component renders correctly 1`] = `
<div>
  Home Page
</div>
`;
